test/boost/mutation_test.cc

```diff
@@ -748,6 +753,16 @@ SEASTAR_TEST_CASE(test_cell_ordering) {
         atomic_cell::make_live(*bytes_type, 1, bytes(), expiry_1, ttl_1),
         atomic_cell::make_live(*bytes_type, 1, bytes(), expiry_2, ttl_1));
 
+    testlog.debug("...then by ttl (in reverse)");
```

Comment:
I don't like these C++ tests because it's not possible to run them against Cassandra and check if we just invented our own rules, or Cassandra also follows the same rules.

Reply:
Maybe we should extend the cql pytest to do that so it can be tested against the reference.
Keep in mind just that the ttl rule is scylla-only, as cassandra stops at the expiry level and if the cells expire at the same time, the one with the higher value (lexicographically) is selected, regardless of their remaining ttl.

Reply:
So we should ask ourselves why we need a different rule. Although I do agree checking the TTL as a penultimate check does make sense - for the same reason that all the cells in a single write will have the same "ttl", not just same expiration. In any case, it's good to have tests that demonstrate how we differ from Cassandra - I don't think you said what you said here (that we won't be identical to Cassandra in this case) in any of the commit messages, for example.

Reply:
It was mentioned, though quite timidly in a57c087c897f60c331d8c9c3abcb125aef9250a2
but it can be expressed more visibly and explicitly tested.

Reply:
In v4, the deviation is documented more clearly in a C++ comment
and a scylla_only pytest tests this